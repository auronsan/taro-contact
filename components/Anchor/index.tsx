import React, { ReactNode } from 'react';
import clsx from 'clsx';
import classes from './anchor.module.css';

type AnchorProps = {
  href: string;
  children: ReactNode;
  // font-size
  size?: 'inherit' | 'xs' | 'sm' | 'md' | 'lg';

  className?: string;
  style?: React.CSSProperties;

  id?: string;
};

export const Anchor: React.FC<AnchorProps> = (props: AnchorProps) => {
  const { href, children, size, className, style, id } = props;
  return (
    <a
      href={href}
      className={clsx(className, { [classes.root]: true, [classes[`fs-${size}`]]: !!size })}
      style={style}
      id={id}
      data-testid={id}
    >
      {children}
    </a>
  );
};
