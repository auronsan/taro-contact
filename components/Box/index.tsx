import { MouseEventHandler } from 'react';

export type BoxProps = {
  /** Class added to the root element, if applicable */
  className?: string;
  /** Inline style added to root component element*/
  style?: React.CSSProperties;
  component?: keyof JSX.IntrinsicElements;
  children?: React.ReactNode;

  // flexbox
  flex?: number;

  // margin
  m?: number;

  // margin-top
  mt?: number;

  // padding
  p?: number;

  // padding-x
  px?: number;

  // padding-y
  py?: number;

  // onclick
  onClick?: MouseEventHandler;

  // role
  role?: string;

  id?: string;
};

export const Box = ({
  children,
  className = '',
  component = 'div',
  flex,
  style,
  m,
  p,
  px,
  py,
  mt,
  role,
  id,
  ...rest
}: BoxProps) => {
  const Component = component;

  return (
    <Component
      {...(className ? { className } : {})}
      role={role}
      id={id}
      data-testid={id}
      style={{
        flex,
        ...style,
        ...(typeof m === 'number' ? { margin: `${m}px` } : {}),
        ...(typeof p === 'number' ? { padding: `${p}px` } : {}),
        ...(typeof px === 'number' ? { paddingLeft: `${px}px`, paddingRight: `${px}px` } : {}),
        ...(typeof py === 'number' ? { paddingTop: `${py}px`, paddingBottom: `${py}px` } : {}),
        ...(typeof mt === 'number' ? { marginTop: `${mt}px` } : {}),
      }}
      {...rest}
    >
      {children}
    </Component>
  );
};
