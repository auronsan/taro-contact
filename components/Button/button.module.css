.root {
  --button-height-sm: rem(40px);
  --button-height: var(--button-height-sm);
  --button-padding-x: var(--spacing-xs);
  --button-fz: var(--font-size-sm);
  --button-radius: var(--radius-sm);

  --button-bd: rem(1px) solid transparent;

  cursor: pointer;
  font-weight: 600;
  position: relative;
  line-height: 1;
  text-align: center;
  overflow: hidden;
  width: auto;
  display: inline-block;

  border-radius: var(--button-radius, var(--radius-default));
  font-size: var(--button-fz, var(--font-size-sm));
  background: var(--button-bg, var(--primary-color));
  border: var(--button-bd, rem(1px) solid transparent);
  color: var(--button-color, var(--color-white));
  height: var(--button-height, var(--button-height-sm));
  padding-inline: var(--button-padding-x, var(--button-padding-x-sm));
  vertical-align: middle;

  &:hover {
    background-color: var(--button-hover);
    color: var(--button-hover-color, var(--button-color));
  }

  &:active {
    transform: scale(0.98);
  }
}

.button-primary {
  --button-bg: var(--primary-color);
  --button-color: var(--color-white);
  --button-hover: var(--primary-color-hover);
  --button-hover-color: var(--color-white);
}

.button-red {
  --button-bg: var(--color-red);
  --button-color: var(--color-white);
  --button-hover: var(--color-red-hover);
  --button-hover-color: var(--color-white);
}

.button-yellow {
  --button-bg: var(--color-yellow);
  --button-color: var(--color-white);
  --button-hover: var(--color-yellow-hover);
  --button-hover-color: var(--color-white);
}
