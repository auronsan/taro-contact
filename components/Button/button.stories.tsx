import React from 'react';
import { Button } from './index';

export default {
  title: 'components/Button',
  component: Button,
};

export const withOnClick = () => <Button onClick={() => {}}>Click Me</Button>;

export const withChildren = () => <Button>Click Me</Button>;

export const withCustomClassName = () => <Button className="custom-class">Click Me</Button>;

export const withRedColor = () => <Button c="red">Click Me</Button>;

export const withOutlineVariant = () => <Button v="outline">Click Me</Button>;

export const withFlexStyle = () => <Button flex={2}>Click Me</Button>;
