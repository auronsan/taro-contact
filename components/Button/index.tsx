import { DOMAttributes } from 'react';
import clsx from 'clsx';
import classes from './button.module.css';

export type ButtonProps = DOMAttributes<HTMLButtonElement> & {
  className?: string;
  children: React.ReactNode;
  // color
  c?: 'primary' | 'red' | 'yellow';

  // variant
  v?: 'outline' | 'filled';

  // flex
  flex?: number;

  style?: React.CSSProperties;

  id?: string;

  type?: 'button' | 'submit';
};

export const Button = ({
  children,
  className,
  c = 'primary',
  v = 'filled',
  flex,
  style,
  id,
  type = 'button',
  ...rest
}: ButtonProps) => (
  <button
    type={type === 'submit' ? 'submit' : 'button'}
    className={clsx(
      classes.root,
      {
        [classes[`button-${c}`]]: !!c,
        [classes[`button-variant-${v}`]]: !!v,
      },
      className
    )}
    style={{
      ...(flex ? { flex } : {}),
      ...style,
    }}
    data-testid={id}
    id={id}
    {...rest}
  >
    {children}
  </button>
);
