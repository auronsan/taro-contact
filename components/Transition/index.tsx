import React from 'react';
import { getTransitionStyles } from './get-transition-styles';
import { TransitionName } from './types';

export type TransitionProps = {
  transition: TransitionName;
  children: (transitionStyles: React.CSSProperties) => React.ReactNode;
  visible?: boolean;
};

export const Transition = (props: TransitionProps) => {
  const { children, transition, visible } = props;

  const transitionStyles = getTransitionStyles({
    transition,
    state: visible ? 'entering' : 'exiting',
    duration: 300,
    timingFunction: 'ease',
  });

  return <>{children(transitionStyles)}</>;
};
