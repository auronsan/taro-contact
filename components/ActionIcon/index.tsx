import { Box } from '@/components/Box';
import classes from './actionIcon.module.css';

type TActionIconProps = {
  onClick?: () => void;
  children?: React.ReactNode;
  id?: string;
};
export const ActionIcon = (props: TActionIconProps): React.ReactElement => {
  const { onClick, children, id } = props;

  return (
    <Box
      className={classes.root}
      onClick={(e) => {
        e.stopPropagation();
        e.preventDefault();
        if (typeof onClick === 'function') {
          onClick();
        }
      }}
      id={id}
      role="button"
    >
      {children}
    </Box>
  );
};
