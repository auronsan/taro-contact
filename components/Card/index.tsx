import React from 'react';
import { Box } from '@/components/Box';
import classes from './card.module.css';

/**
 * Props for the Card component.
 */
type TCardProps = {
  /** The content to be rendered inside the card. */
  children: React.ReactNode;
};

/**
 * Card component.
 * @param props - The props for the component.
 * @returns The rendered card.
 */
export const Card = ({ children }: TCardProps): React.ReactElement => (
  <Box className={classes.card}>
    <Box>{children}</Box>
  </Box>
);
