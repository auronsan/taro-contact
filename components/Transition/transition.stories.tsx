import React from 'react';
import { Button } from '@/components/Button';
import { Transition } from './index';

export default {
  title: 'components/Transition',
  component: Transition,
};

export const Default = () => {
  const [visible, setVisible] = React.useState(false);

  return (
    <>
      <Button onClick={() => setVisible(!visible)}>Toggle Transition</Button>
      <Transition transition="fade" visible={visible}>
        {(transitionStyles) => (
          <div style={transitionStyles}>
            <h1>Transition Component</h1>
          </div>
        )}
      </Transition>
    </>
  );
};
