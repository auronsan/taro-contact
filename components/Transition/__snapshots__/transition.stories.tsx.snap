// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Transition Default smoke-test 1`] = `
<main class="roboto-normal">
  <button type="button"
          class="button_root__s7zfG button_button-primary__cWRWo button_button-variant-filled__Dz0lC"
  >
    Toggle Transition
  </button>
  <div style="transition-property: opacity; transition-duration: 300ms; transition-timing-function: ease; opacity: 0;">
    <h1>
      Transition Component
    </h1>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
